packages/taquito/src/contract/rpc-contract-provider.ts

```diff
@@ -395,6 +400,127 @@ export class RpcContractProvider extends Provider implements ContractProvider, S
     return new TransactionOperation(hash, content, source, forgedBytes, opResponse, context);
   }
 
+  /**
+   *
+   * @description Stake tz from current address to a specific address. Built on top of the existing transaction operation
+   *
+   * @returns An operation handle with the result from the rpc node
+   *
+   * @param Stake pseudo-operation parameter
+   */
+  async stake(params: StakeParams) {
+    if (params.to === undefined) {
+      params.to = params.source;
```

Comment:
might be ideal to `sourceValidation` first then assign it to `params.to` (same apply to other two pseudo ops)